fc/protocol.py

```diff
@@ -235,12 +240,46 @@ def _parse(self):
 
         # Store unit definitions
         for units in details.get('units', []):
+            Protocol.units_list.append(units)
             self.units.add_unit(units.name, units.pint_expression)
 
+        # check simulation has a model interface
+        def has_model_interface(simulation):
+            """ Check wether the simulation has a model interface """
+            try:
+                simulation.nested_sim.model.proto.model_interface
+            except AttributeError:
+                return False
+            else:
+                return True
+
         # Create model interface
-        def process_interface(interface):
+        def process_interface(interface, simulations):
             """ Process a protocol's model interface. """
-            self.model_interface = interface  # TODO: Merging!
+            self.model_interface = interface
+            # for any nested simulations merge the model interface from
+            # the nested simulation to this outer model interface
+            for simulation in simulations:
+                if isinstance(simulation, fc.simulations.simulations.Nested):
+                    # if the simulation has its own protocol add units
+                    # and merge model interface
+                    if has_model_interface(simulation):
+                        self.model_interface.merge(simulation.nested_sim.model.proto.model_interface)
+
+                        # need to add units from inner protocol
+
+                        # this doesnt work as the unit store names the units from
```

Comment:
I see the difficulty, and yes, lots of the code in `actions.py` assumes you can look up units by name. While for `VariableReference` instances we could do all lookups as a pre-processing step using the correct protocol's `UnitStore`, this wouldn't be easy for numbers in equations. And we'd need to worry about making nested protocol unit stores use the same underlying registry too!

It might be better for `Protocol` to store `self.unit_definitions = details.get('units', [])`. We can then merge reliably (I think) with
```python
    nested_proto = simulation.nested_sim.model.proto
    self.unit_definitions.extend(nested_proto.unit_definitions)
    for units in nested_proto.unit_definitions:
        self.units.add_unit(units.name, units.pint_expression)
```